### DISCUSSION TITLE
Can Dify parse images in pdfs and embed image into vector databases currently?

### DISCUSSION THREAD

**Discussion by taowang1993** [langgenius/dify#8654] [Help]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

Can Dify parse images in pdfs and embed image into vector databases currently?

I have a pdf book which has many images and tables. 

My dify is self hostes with the Unstructured instance.

**Comment by seungwoos** [comment]
So far, I guess dify does not support image and table parsing; you should implement your own pdf parser module. You should implement your own extractor using Unstructured's `partition_pdf`.

**Reply by taowang1993** [reply]
So I need to add an "unstructured_pdf_extractor.py" file inside the unstructured folder, right?

But dify already has a pdf_extractor which only extracts text at the current time. 

I am afraid these two files will confict.

<img width="317" alt="image" src="https://github.com/user-attachments/assets/6958300e-ea5f-472f-9eb6-f9457f402eb1">

**Reply by seungwoos** [reply]
@taowang1993 

You should change this part when you set `ETL_TYPE=Unstructured` in `.env`.
https://github.com/langgenius/dify/blob/7f1b0288409626fcb87bee2fe12643ebfa0f3059/api/core/rag/extractor/extract_processor.py#L104
